static-assets/components/cstudio-forms/forms-engine.js

```diff
@@ -2598,7 +2598,15 @@ var CStudioForms =
               }
 
               const defaultValue = moduleConfig.config.field.defaultValue;
-              if (!value && defaultValue && typeof defaultValue === 'string' && !noDefaultLookup[formField.id]) {
+              // When value is of type string, an empty string is considered unset (unless it is contained in the
+              // `noDefaultLookup`, where it means that an empty string was the value set).
+              const unsetStoredValue = typeof value === 'string' ? !value : craftercms.utils.object.nou(value);
```

Comment:
@rart I was wrong about the `no-default` attribute in boolean values. It's only for string values as we have no distinction between empty and not set (as they end up being an empty string). For boolean, the stored value will never be an empty string. So, the issue was that when rendering the field, if the value was set to false by the control, when re-rendering the repeating group it was going to be considered as 'false' in the condition and ended up setting the defaultValue.